internal/archive/archive.go

```diff
@@ -129,6 +138,44 @@ func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, error) {
 const ubuntuURL = "http://archive.ubuntu.com/ubuntu/"
 const ubuntuPortsURL = "http://ports.ubuntu.com/ubuntu-ports/"
 
+var proArchiveInfo = map[string]struct {
+	baseURL, label string
+}{
+	"fips": {
+		baseURL: "https://esm.ubuntu.com/fips/ubuntu/",
+		label:   "UbuntuFIPS",
+	},
+	"fips-updates": {
+		baseURL: "https://esm.ubuntu.com/fips-updates/ubuntu/",
+		label:   "UbuntuFIPSUpdates",
+	},
+	"apps": {
+		baseURL: "https://esm.ubuntu.com/apps/ubuntu/",
+		label:   "UbuntuESMApps",
+	},
+	"infra": {
+		baseURL: "https://esm.ubuntu.com/infra/ubuntu/",
+		label:   "UbuntuESM",
+	},
+}
+
+func archiveURL(pro, arch string) string {
+	if pro == "" {
+		if arch == "amd64" || arch == "i386" {
```

Comment:
Question: Is this list fixed, at least for the near future? Bear in mind that if it changes then we need to release a new version of Chisel instead of changing the yaml.

The changes here would not be in this PR, but I think we need to have that discussion.

Reply:
For non-pro archives, it does seem that way. For `amd64` and `i386`, the `archive.ubuntu.com..` is the one to go and for others the `ports.ubuntu.com` is the one.

This comment does, however, raise a good point about the `pro` archives though. Not all the `pro` archives have support for all the architectures. I wonder what we should do if an user tries to `cut` with an architecture that the highest priority pro archive does not have packages for.

Reply:
Agreed on three things:
* We will change this function to have an early return.
* Look into a way of programmatically find what architectures are in ubuntu.com or in ports.ubuntu.com because the InRelease file contains all of them (both in the header and in the listed paths).
* If one of the archives (fips or not) does not have an architecture we should fall through and emit a warning, not an error when opening it but emit a warning. Not decided, we need more data.

Reply:
One more thing:
* Keep constants in setup and map in archives and we do not need to create another package for now.
* Remove the None constant if using string.

Reply:
No. In most cases, either the package name or the package version contains `fips`. However, there are some exceptions, e.g. for focal fips, the `shim`, `shim-dbg` and `shim-signed` contain `fips` in neither the package names nor the versions.